satpy/readers/seviri_l1b_nc.py

```diff
@@ -60,29 +61,46 @@ def end_time(self):
         """Get the end time."""
         return self.deltaEnd
 
-    def _read_file(self):
+    @cached_property
+    def nc(self):
         """Read the file."""
-        if self.nc is None:
+        f_obj = open_file_or_filename(self.filename)
 
-            self.nc = xr.open_dataset(self.filename,
-                                      decode_cf=True,
-                                      mask_and_scale=False,
-                                      chunks=CHUNK_SIZE)
+        dataset = xr.open_dataset(f_obj,
+                                  decode_cf=True,
+                                  mask_and_scale=False,
+                                  chunks=CHUNK_SIZE)
```

Comment:
This pattern is being used in multiple places now, and it might need to evolve in the future if fsspec starts return PathLib compliant files or if xarray starts supporting s3buckets for example. I think it would be beneficial if we could create a reader agnostic function that would do this, so in the future if the APIs of dependencies change, it would make it easier to change. So how about a function along these lines:
```python
def open_dataset(filename, *args,  **kwargs):
    f_obj = open_file_or_filename(filename)
    return xr.open_dataset(f_obj, *args, **kwargs)
```
This goes a bit against my conviction that we should avoid *args and **kwargs though... maybe a decorator would be an alternative?

Reply:
Do I understand correctly that you want to move the logic into a function completely outside of the reader so it can be used in other readers and the `nc` method just calls `open_dataset`?

Reply:
Yes exactly.

Reply:
Ok. So where would you think that function should go (maybe in file_handlers.py)?

Reply:
Sounds about right!